src/server.cpp

```diff
@@ -1566,13 +1566,41 @@ void Server::SendShowFormspecMessage(session_t peer_id, const std::string &forms
                                      const std::string &formname)
 {
 	NetworkPacket pkt(TOCLIENT_SHOW_FORMSPEC, 0 , peer_id);
+	std::string client_name;
+
+	auto peer_state_iterator = m_formspec_state_data.find(peer_id);
 	if (formspec.empty()){
 		//the client should close the formspec
-		pkt.putLongString("");
+		if (peer_state_iterator != m_formspec_state_data.end()) {
+			PeerFormspecState peer_state = peer_state_iterator->second;
+			if (!formname.empty()) {
+				if (peer_state.formspec_name != formname) {
+					errorstream << "attempting to close formspec '" << formname
```

Comment:
Shouldn't be `errorsteam`, as the client will also close the formspec with an empty formspec name (see `void Game::handleClientEvent_ShowFormSpec`)

Reply:
Seems strange to have a mod attempting to close a formspec without opening it first, but I suppose strictly speaking that doesn't violate the current API, it's just a no-op so I will downgrade it to warningstream